### PR TITLE
feat: extend `useTable.setFilters` function

### PR DESCRIPTION
**Pull request by aliemir**
Currently, `setFilters` function returned from the `useTable` hook (package: `@pankod/refine-core`) works with `merge` behavior. meaning; when you pass an array of `CrudFilter`s, it will append the new filters to the existing ones and remove the duplicates. 

We're introducing a second parameter to the `setFilters` function: `behavior`. It's default to `merge` but can also be `replace` which might be the preferred behavior for some use cases. 

**Usage**
```ts
setFilters(newFilters: CrudFilters, behavior: "merge" | "replace")
```

Default behavior can be set through `useTable` parameters with `defaultSetFilterBehavior`

**Usage**
```ts
const { /* */ } = useTable({
  /* ... */
  defaultSetFilterBehavior: "replace",
  /* ... */
}); 
```

We're also including an overload for the `setFilters` function as a setter function. 
**Usage**
```ts
setFilters((prevFilters: CrudFilters) => CrudFilters)
```

**Status**

- [x] feat: `merge` and `replace` behaviors added
- [x] feat: setter function overload added
- [x] test: add new test cases for the new behaviors
- [x] bug: remove filter is not working properly on `merge`
- [x] docs: `useTable` docs needs to be updated

### CODE REVIEW

File: .changeset/forty-humans-carry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@pankod/refine-mui": patch
```
Comment (aliemir) [inline .changeset/forty-humans-carry.md:2]:
```suggestion
"@pankod/refine-mui": minor
```

File: .changeset/moody-pandas-pay.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@pankod/refine-antd": patch
```
Comment (aliemir) [inline .changeset/moody-pandas-pay.md:2]:
```suggestion
"@pankod/refine-antd": minor
```

File: documentation/docs/core/hooks/useTable.md [hunk 3]
```diff
@@ -154,28 +158,73 @@ setFilter([
 ]);
 
 console.log(filters); // [{ field: "title", operator: "contains", value: "A" }, { field: "status", operator: "equals", value: "published" }]
+
+setFilter([
+    {
+        field: "author",
+        operator: "contains",
+        value: "Ali",
```
Comment (omeraplak) [inline documentation/docs/core/hooks/useTable.md:166]:
```suggestion
        value: "Foo",
```

File: packages/antd/src/hooks/table/useTable/useTable.ts [hunk 4]
```diff
@@ -97,6 +98,7 @@ export const useTable = <
         initialCurrent,
         initialPageSize,
         initialSorter,
+        defaultSetFilterBehavior,
```
Comment (salihozdemir) [inline packages/antd/src/hooks/table/useTable/useTable.ts:101]: `useTable` hook doesn't export the `setFilters`, `setSorters` etc. in `@pankod/refine-antd` package so `defaultSetFilterBehavior` doesn't create a use case  🤔

Comment (aliemir) [inline packages/antd/src/hooks/table/useTable/useTable.ts:101]:
```suggestion
```

**Review by salihozdemir — APPROVED** [review]

**Pull request merged**